src/test/coll.rs

```diff
@@ -545,3 +546,91 @@ async fn ns_not_found_suppression() {
     coll.drop(None).await.expect("drop should not fail");
     coll.drop(None).await.expect("drop should not fail");
 }
+
+async fn delete_hint_test(options: Option<DeleteOptions>, name: &str) {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = EventClient::new().await;
+    let coll = client.database(name).collection(name);
+    let _ = coll.delete_many(doc! {}, options.clone()).await;
```

Comment:
Patrick and I have started doing `let _: Result =` instead of `let _ =`. Otherwise, we might accidentally ignore something besides a result that's `must_use`, e.g. forgetting to await a future.